ai_chatbots/chatbots.py

```diff
@@ -82,21 +84,16 @@ def create_tools(self):
     def get_llm(self, **kwargs) -> BaseChatModel:
         """
         Return the LLM instance for the chatbot.
-        Determine the LLM class to use based on the AI_PROVIDER setting.
         Set it up to use a proxy, with required proxy kwargs, if applicable.
         Bind the LLM to any tools if they are present.
         """
-        try:
-            llm_class = LLMClassEnum[settings.AI_PROVIDER].value
-        except KeyError:
-            raise NotImplementedError from KeyError
-        llm = llm_class(
-            model=self.model,
+        llm = ChatLiteLLM(
+            model=f"{self.model_prefix}{self.model}",
             **(self.proxy.get_api_kwargs() if self.proxy else {}),
             **(self.proxy.get_additional_kwargs(self) if self.proxy else {}),
             **kwargs,
         )
-        if self.temperature:
+        if self.temperature and self.model not in settings.AI_UNSUPPORTED_TEMP_MODELS:
```

Comment:
The new `o3-mini` model does not support the temperature parameter and will raise an exception if it is passed along